### DISCUSSION TITLE
I need ...

### DISCUSSION THREAD

**Discussion by DiTsi** [eslupmi/impulse#193] [Ideas]
If you need some feature, please describe it here

**Comment by slidingshade** [comment]
Would be great to have all features with chains and user assigning not only from buttons in messengers but from UI + user authentication  for UI.

**Reply by DiTsi** [reply]
Yeah, authorization and UI buttons are planned. But what do you mean by 'features with chain'?

**Reply by slidingshade** [reply]
I mean just to do "Take" and "Realease" stuff like from in-message buttons but in UI.
Also some kind of integration with Jira would be cool. Something like a button to create Jira ticket from  incident and the showing a link to it.

**Reply by DiTsi** [reply]
Yeah, we’re planning all the features you mentioned!
But before adding buttons to the UI, we need to implement authentication.

All these features are expected to take about 2–3 months.

**Reply by DiTsi** [reply]
Hi @slidingshade. We plan features you mention and we interested why you need user auth for UI? For what?

**Reply by DiTsi** [reply]
Twice longer but we did it. See v3.5.0

**Comment by slidingshade** [comment]
Just to use UI in a standalone mode, without buttons in messages (maybe even disable in-message buttons) So you can open the dashboard and for example take incident to work and it would be assigned to authenticated. \
Btw, I saw you've implemented Jira support in latest release, does it work for on-prem Jira or only for cloud?

**Reply by DiTsi** [reply]
Okay, I understand, thank you.

We tested Jira integration on cloud version. We don't have an on-prem, and can't test it. Do it please, if you have one. We will be appreciated

**Comment by maartenberg** [comment]
Hi!

We (cc @Riscky) would like to be able to use IMPulse in HA mode without having to use NFS to share a filesystem.
To this end we'd like to propose adding support for storing the data_dir in S3 (or a compatible storage) as opposed to on the local filesystem.
We are willing to make the PRs, if you are okay with merging them.

Our implementation plan for making this possible would consist of going over all cases where IMPulse writes to the data_dir and making this use an intermediate class instead, then make this class swappable for a different implementation.
This way it would also become easier to add support for other storage backends later on if needed.

Separately of the data storage we'd also like to add support for coordinating the state lock via Consul instead of storing files in the data_dir, we have a similar plan for supporting this.

Would a PR / set of PRs to add optional support for these features be acceptable?

**Reply by DiTsi** [reply]
Hi, @maartenberg! We considered implementing data storage in S3, but didn't because it wasn't necessary.
Thank you for your willingness to participate! We'll likely close the contribution (license issues), but we're happy to accept ideas.

We'll look into how to implement it technically, and I'll get back to you with an answer by September 15th.

**Reply by DiTsi** [reply]
Hi! We'll implement S3 support in `v3.8.0`.
Track corresponding [issue](https://github.com/eslupmi/impulse/issues/588) or subscribe to IMPulse releases
